api/all/src/main/java/io/opentelemetry/api/internal/ReadOnlyArrayMap.java

```diff
@@ -32,7 +32,7 @@
 
 /** A read-only view of an array of key-value pairs. */
 @SuppressWarnings("unchecked")
-public final class ReadOnlyArrayMap<K, V> implements Map<K, V> {
+public final class ReadOnlyArrayMap<K, V> extends AbstractMap<K, V> {
```

Comment:
this gets us valid Map equals/hashcode, which we didn't have before.

Reply:
Should this be sent as a standalone change anyway?

Reply:
yes, I was thinking that as well. will do, for sure.

Reply:
https://github.com/open-telemetry/opentelemetry-java/pull/2963